### ISSUE TITLE
Automated texture fix corrupts the final code

### ISSUE THREAD

**Issue by dunquake** [vyruss-zz/aneurysmpublic#1]
Opening a target ROM with aneurysm and selecting the option to patch the corrupted textures (which by default the program seems to recognize) will utterly corrupt the game's code. It becomes playable for only a very short amount of time and has noticeable glitches when moving (mostly texture pop-in). It hard crashes a couple of minutes after playing. Visualizing the ROM through a hex editor suggests the patching feature simply replicates textures far longer than it needs to.

Additionally, several emulators tested with the patched ROM will quickly point out that the checksum is incorrect, so either a third-party tool - or an emulator with automated checksum fix - is required.

Is aneurysm looking for a particular ROM version or any particular ROM checksum?

Please advise.

**Comment by vyruss-zz** [comment]
Blood Shot (UE) [!], CRC32 F9F2BCEB
Original SHA1 hash 513005efd123539a905986130d15125085837559.

If you have your original file information available I'd like to have it, I am not sure of other versions being around.

Some diagnostic questions,

Did you run from source in Eclipse or execute the jar included with the repo?

What OS are you using?

This was tested on a real Genesis as well as with Bizhawk and Exodus when it was put together, so I am deeply interested in what is different.

**Comment by dunquake** [comment]
Hi, thank you for getting back to me.

The CRC32 and SHA1 match with the ROM I have; the only thing I had to manually change was the file's extension from *.md to *bin so aneurysm would recognize it. I'm using the jar version readily available from the repository, on Windows 10 64-bit. The emulators I've used were Bizhawk 2.3, Fusion 3.64, and Gens KMod v0.7.3.7.

However, I should point out that I've just tried using aneurysm again and this time it worked. On the first run of the jar, it detected the ROM as having the corrupted textures; I chose to apply the patch, and the program closed after that. I then executed the jar again and it gave me no warnings - I thought everything was alright, as aneurysm then seemed to be working fine. It was after I tested the ROM that problems came up.

But now I chose to patch the ROM and aneurysm didn't close, it displayed its interface right after. And the ROM is working now, textures fixed. No issues with the emulators I've mentioned. I have to assume some error occurred on the first run.

Thank you for your help!

PS: I also made the initial comment on the Doomworld forums - since it's solved, I'll correct it.
